service/history/workflowTaskHandler.go

```diff
@@ -568,9 +569,7 @@ func (handler *workflowTaskHandlerImpl) handleCommandCompleteWorkflow(
 		return handler.failWorkflowTask(enumspb.WORKFLOW_TASK_FAILED_CAUSE_UNHANDLED_COMMAND, nil)
 	}
 
-	if handler.updateRegistry.HasUndeliveredUpdates() {
-		return handler.failWorkflowTask(enumspb.WORKFLOW_TASK_FAILED_CAUSE_UNHANDLED_UPDATE, nil)
```

Comment:
So did we have a requirements change? Undelivered updates no longer prevent completion?

Reply:
Yes, fail to close has huge cost because it has to replay the whole history, this is especially bad when it has high frequency signal/update that prevent workflow from closing.